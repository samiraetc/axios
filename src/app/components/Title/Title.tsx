import React, {useState, useEffect} from 'react'
import { useFruits } from '../../hook/useFruits';

const Title = () => {
  const {pokemon: poke, loading} = useFruits({resource: 'api/v2/pokemon/ditto'})

  console.log(poke);

  if (loading) {
    return <p>Loading...</p>;
  }

  return (
    <div>Ol√° mundo</div>
  )
}

export default Title