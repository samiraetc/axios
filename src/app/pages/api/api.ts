
import axios from 'axios';

export const baseURL =  'https://pokeapi.co/'

const api = axios.create({
  baseURL
});

export default api;