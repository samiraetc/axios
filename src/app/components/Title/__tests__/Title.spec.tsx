import { render, cleanup, waitFor } from '@testing-library/react';
import React from 'react';
import Title from '../Title';
import MockAdapter from "axios-mock-adapter";
import api, { baseURL } from "@pages/api/api";
import { titleMock } from '../../../mock/titleMock';

const mock = new MockAdapter(api, { onNoMatch: "throwException" });

beforeAll(() => {
  mock.reset();
});

afterEach(cleanup);


describe('<Title />', () => {
  it('should render', async () => {
    mock.onGet(`${baseURL}api/v2/pokemon/ditto`).reply(200, titleMock);
    const { asFragment, getByText, queryByText } = render(
     <Title />
    );

    expect(queryByText('Loading...')).toBeInTheDocument();
   

    await waitFor(() =>  expect(queryByText('Loading...')).not.toBeInTheDocument());
   
    expect(getByText('Olá mundo')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render and api 404', async () => {

    mock.onGet(`${baseURL}api/v2/pokemon/ditto`).reply(404, {errors: 'volte a tentar novamente'});

    const { asFragment, getByText, queryByText } = render(
     <Title />
    );

    expect(queryByText('Loading...')).toBeInTheDocument();
 
    await waitFor(() =>  expect(queryByText('Loading...')).not.toBeInTheDocument());
   
    expect(getByText('Olá mundo')).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
