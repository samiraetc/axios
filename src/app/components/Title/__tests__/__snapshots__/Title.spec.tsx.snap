// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> should render 1`] = `
<DocumentFragment>
  <div>
    Olá mundo
  </div>
</DocumentFragment>
`;

exports[`<Title /> should render and api 404 1`] = `
<DocumentFragment>
  <div>
    Olá mundo
  </div>
</DocumentFragment>
`;
